src/hooks/core/onesignal/useOneSignalSubscribeToCommunityHooks.ts

```diff
@@ -26,12 +63,73 @@ const useOneSignalSubscribeToCommunityHooks = () => {
 
   const addNotificationListener = async () => {
     OneSignal.Notifications.addEventListener('click', (event) => {
-      const additionalData = event?.notification?.additionalData as OneSignalTopicNotificationData;
+      const additionalData = event?.notification?.additionalData as OneSignalNotificationData;
+
       if (additionalData?.community) {
-        navigation.navigate('TopicPageScreen', {
-          id: additionalData?.community
+        AnalyticsEventTracking.eventTrack(
+          `${BetterSocialEventTracking.ONE_SIGNAL_PUSH_NOTIFICATIONS}${
+            additionalData?.campaign || 'topic'
+          }`
+        );
+        baseResetNavigation('Feed', {
+          screen: 'TopicPageScreen',
+          params: {
+            id: additionalData?.community,
+            is_from_pn: true
+          }
+        });
+
+        return;
+      }
+
+      // This block is for handling notification from one signal
+      if (additionalData?.type === ONESIGNAL_PN_TYPE.PDP) {
+        AnalyticsEventTracking.eventTrack(
+          `${BetterSocialEventTracking.ONE_SIGNAL_PUSH_NOTIFICATIONS}${additionalData?.campaign}`
+        );
+
+        baseResetNavigation('Feed', {
+          screen: 'PostDetailPage',
+          params: {
+            feedId: additionalData?.feed_id,
+            is_from_pn: true
+          }
+        });
+
+        return;
+      }
+
+      if (additionalData?.type === ONESIGNAL_PN_TYPE.TOPIC) {
+        AnalyticsEventTracking.eventTrack(
+          `${BetterSocialEventTracking.ONE_SIGNAL_PUSH_NOTIFICATIONS}${additionalData?.campaign}`
+        );
+        baseResetNavigation('Feed', {
+          screen: 'TopicPageScreen',
+          params: {
+            id: additionalData?.community,
+            is_from_pn: true
+          }
+        });
+
+        return;
+      }
+
+      if (additionalData.type === ONESIGNAL_PN_TYPE.PROFILE) {
+        AnalyticsEventTracking.eventTrack(
+          `${BetterSocialEventTracking.ONE_SIGNAL_PUSH_NOTIFICATIONS}${additionalData?.campaign}`
+        );
+        baseResetNavigation('Feed', {
+          screen: 'OtherProfile',
+          params: {
+            data: {
+              user_id: signedProfileId,
+              username: additionalData?.username,
+              is_from_pn: true
+            }
+          }
         });
       }
+      // Onesignal block end
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for handling different types of notifications seems repetitive. The same pattern of tracking an event and resetting navigation is repeated for each type of notification. Consider refactoring this into a separate function that takes the necessary parameters and performs these actions. This will make the code more DRY (Don't Repeat Yourself) and easier to maintain.

```diff
-      if (additionalData?.type === ONESIGNAL_PN_TYPE.PDP) {
-        AnalyticsEventTracking.eventTrack(
-          `${BetterSocialEventTracking.ONE_SIGNAL_PUSH_NOTIFICATIONS}${additionalData?.campaign}`
-        );
-
-        baseResetNavigation('Feed', {
-          screen: 'PostDetailPage',
-          params: {
-            feedId: additionalData?.feed_id,
-            is_from_pn: true
-          }
-        });
-
-        return;
-      }
-
-      if (additionalData?.type === ONESIGNAL_PN_TYPE.TOPIC) {
-        AnalyticsEventTracking.eventTrack(
-          `${BetterSocialEventTracking.ONE_SIGNAL_PUSH_NOTIFICATIONS}${additionalData?.campaign}`
-        );
-        baseResetNavigation('Feed', {
-          screen: 'TopicPageScreen',
-          params: {
-            id: additionalData?.community,
-            is_from_pn: true
-          }
-        });
-
-        return;
-      }
-
-      if (additionalData.type === ONESIGNAL_PN_TYPE.PROFILE) {
-        AnalyticsEventTracking.eventTrack(
-          `${BetterSocialEventTracking.ONE_SIGNAL_PUSH_NOTIFICATIONS}${additionalData?.campaign}`
-        );
-        baseResetNavigation('Feed', {
-          screen: 'OtherProfile',
-          params: {
-            data: {
-              user_id: signedProfileId,
-              username: additionalData?.username,
-              is_from_pn: true
-            }
-          }
-         });
-       }
+      handleNotification(additionalData);
```

In the `handleNotification` function, you can then implement the repetitive logic:

```javascript
const handleNotification = (additionalData) => {
  let screenName;
  let params;

  switch (additionalData?.type) {
    case ONESIGNAL_PN_TYPE.PDP:
      screenName = 'PostDetailPage';
      params = {
        feedId: additionalData?.feed_id,
        is_from_pn: true
      };
      break;
    case ONESIGNAL_PN_TYPE.TOPIC:
      screenName = 'TopicPageScreen';
      params = {
        id: additionalData?.community,
        is_from_pn: true
      };
      break;
    case ONESIGNAL_PN_TYPE.PROFILE:
      screenName = 'OtherProfile';
      params = {
        data: {
          user_id: signedProfileId,
          username: additionalData?.username,
          is_from_pn: true
        }
      };
      break;
    default:
      return;
  }

  AnalyticsEventTracking.eventTrack(
    `${BetterSocialEventTracking.ONE_SIGNAL_PUSH_NOTIFICATIONS}${additionalData?.campaign}`
  );

  baseResetNavigation('Feed', {
    screen: screenName,
    params
  });
};
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->